### ISSUE TITLE
Problem Unpacking Solution using PP 2.0 Tools

### ISSUE THREAD

**Issue by cfguy** [microsoft/powerplatform-build-tools#204]
Hi folks - I'm having issues all of the sudden with unpacking my solution. A couple of oddities: this use to work without any issue and now for all of my pipelines using similar code, it's having an issue on the unpack step finding the file; additionally, this step works fine when using v0.0 of the task. First time Git user, so please talk to me like I'm a 3rd grader. Appreciate any help!

![2022-09-16_09-05-59](https://user-images.githubusercontent.com/113710787/190685407-9bc01e75-a1df-453a-8a3e-3b7386a428ee.png)

![2022-09-16_09-05-02](https://user-images.githubusercontent.com/113710787/190685965-2b8e5387-949a-4456-b2a7-7e694edf8d23.png)

**Comment by davidjenni** [comment]
@cfguy your screenshots do help to see what's going on: your unpack-solution step fails since its SolutionFile input property doesn't agree with the actual name of the exported solution.zip:

When exporting a solution, the underlying pac CLI enforces different names for the managed and unmanaged zip file to avoid clobbering the local file: When exporting with `--managed`, the solution's base name is always being appended with `_managed.zip`. In your example/pipeline, this is what happens:
```
 pac solution export --name QLSWASUpdateManaged --path out\tmp\QLSWASUpdateManaged.zip --managed    
```
results in the file stored as:
```
out\tmp\QLSWASUpdateManaged_managed.zip 
```

So to fix your unpack task's failure, change your task's input for the Solution FIle to:
![image](https://user-images.githubusercontent.com/3200210/190709814-4cbfc1cd-12c2-4b91-b60f-20a4e2f40d0a.png)

I'm filing an internal bug for us to have the actual solution file that is written to disk to also show up in the log (which would have helped you to discover that on your own)

AB#2925297

**Comment by cfguy** [comment]
Well it appears that I've come to the right place for answers. That did resolve my issue, and yes, I think adding some output details to the log would have allowed me to figure that out myself. 

This must have been a new change with release 2.07 as it worked without issue just yesterday morning (without _managed extension) in version 2.05 which is curious.

**Comment by mikefactorial** [comment]
Ran across the same issue today. Previously, the task would honor the name given without appending "_managed". This changed with the latest update.

**Comment by davegoopot** [comment]
Is the plan to leave this as is now or revert to the previous behaviour?

The behaviour change has broken a bunch of automation we had already in place.

Why was it changed?

**Comment by davidjenni** [comment]
I did some src code/git blame spelunking in our pac CLI repo... seems like a classical "no good deed goes unpunished" where we refactored code and addressed different but related bugs.
I'll revert to the old behavior for pac solution export to always use the full file path.

**Comment by YannickRe** [comment]
Might I suggest some unit tests to avoid introducing backwards incompatible "fixes"? Or a quicker release schedule (or hotfix release)?
It's pretty inconvenient to have my pipelines fail now, due to an automated update of these tasks, which I will have to address now, and with the next release I have to undo my workaround because you'll have restored the previous behavior.

And I get that stuff happens, but you are building tooling that you are pushing as the defacto standard. You can't have too many of these happening before people lose faith in the stability of this approach (and that would be a shame).
I mean this in the most well-intended way ;)

**Comment by davidjenni** [comment]
Fixed with 2.0.8; note that due to Azure DevOps's caching, it might take several hours until all build agents run with the latest update

https://github.com/microsoft/powerplatform-build-tools/releases/tag/v2.0.8

CC: @cfguy @davegoopot @mikefactorial
